gocse/schema/json_schema.go

```diff
@@ -55,7 +54,7 @@ func CreateJSONSchema(dataKeyBase64 string) bson.Raw {
 	schema := fmt.Sprintf(jsc, dataKeyBase64)
 	var schemaDoc bson.Raw
 	if err := bson.UnmarshalExtJSON([]byte(schema), true, &schemaDoc); err != nil {
-		log.Fatalf("UnmarshalExtJSON error: %v", err)
+		return nil, fmt.Errorf("UnmarshalExtJSON error: %v", err)
```

Comment:
Does `fmt.Errorf` return a new instance of `error` or should you use `new Error()` to wrap it here?

Reply:
I guess it's the former.

Reply:
If error is nil then it makes a new error, otherwise it wraps it. I'm unsure what the importance of this is here.

https://golang.org/src/fmt/errors.go?s=624:674#L7

Reply:
Sorry, I answered my own question in the second comment.